### ISSUE TITLE
When passing dataframe to geopandas.GeoDataFrame the original frame is altered.

### ISSUE THREAD

**Issue by BenScarrott** [geopandas/geopandas#1804]
I think this is unexpected behaviour? 
I would expect the original frame to remain unaltered.

```py
import urllib.request
import pandas as pd
import geopandas as gpd

urllib.request.urlretrieve('https://energydata.info/dataset/f85d1796-e7f2-4630-be84-79420174e3bd/resource/6e640a13-cab4-457b-b9e6-0336051bac27/download/healthmopupandbaselinenmisfacility.csv', 'healthmopupandbaselinenmisfacility.csv')

hospital_data = pd.read_csv('healthmopupandbaselinenmisfacility.csv')

print(len(hospital_data.columns))

hosp_gdf = gpd.GeoDataFrame(hospital_data, geometry=gpd.points_from_xy(hospital_data.longitude, hospital_data.latitude))
hosp_gdf.crs = "EPSG:4326"

print(len(hospital_data.columns))
```

**Comment by martinfleis** [comment]
Thanks for the nice report!

Yeah, this is a known bug reported in #1179 with a proposal on how to fix it in #1324 but we haven't got there yet.

I'll close it as a duplication of an existing (old!) issue.

**Comment by BenScarrott** [comment]
Thanks Martin, apologies for the duplication!
